### DISCUSSION TITLE
Why subtract microseconds for RowParser

### DISCUSSION THREAD

**Discussion by petmo** [gbeced/basana#19] [Q&A] [answered]
First off, love this package. 

Trying to replicate results in live view vs backtest view by aligning the data. In the rowparser, we're subtracing a microsecond for each bar event and I'm struggling to understand why - could anyone help shed some light?

<img width="879" alt="Screenshot 2024-01-18 at 20 47 49" src="https://github.com/gbeced/basana/assets/26343993/f7d9ddd3-8d4f-4009-aae3-cd8565dc0db7">

**Comment by gbeced** [comment, marked as answer]
It was a bad decision. The rationale behind that is that the trading summary of day X gets reported in the form of a bar event right at the end of day X.
I now realize it it was a bad decision because for 24*7 markets you could have a trade in the last microsecond and that should be included in the trading summary of day X, whose bar event should get reported at X + 1.
I'll fix that for the next version.

**Reply by gbeced** [reply]
This is now fixed in basana-1.5.1.
